ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/validator/ExpressionValidator.java

```diff
@@ -42,20 +43,15 @@ public void setNextValidator(Validator nextValidator) {
     @Override
     public boolean evaluate(String source) {
         IncompleteInputFinder incompleteInputFinder = new IncompleteInputFinder();
-        boolean isIncomplete = NodeParser.parseExpression(source).apply(incompleteInputFinder);
-        if (!NodeParser.parseExpression(source).hasDiagnostics()) {
-            return false;
-        } else {
-            if (!source.endsWith(";") && !NodeParser.parseExpression(source + ";").hasDiagnostics()) {
-                return false;
-            }
+        Node parsedNode = NodeParser.parseExpression(source);
+        if (!parsedNode.hasDiagnostics()) {
+            return true;
         }
 
-        if (!isIncomplete) {
-            return nextInValidator.evaluate(source);
-        } else {
+        if (!parsedNode.apply(incompleteInputFinder)) {
             return true;
         }
 
+        return nextInValidator.evaluate(source);
```

Comment:
We can merge these together

Reply:
Updated in 7d25e1dea188151fc8c5f84a1bf9156ce22ebc81